lib/internal/bootstrap/pre_execution.js

```diff
@@ -293,6 +293,24 @@ function initializeDeprecations() {
     process._tickCallback = deprecate(process._tickCallback,
                                       'process._tickCallback() is deprecated',
                                       'DEP0134');
+
+    process.env = new Proxy(process.env, {
+      set: function(obj, prop, value) {
+        const nonValidType = (typeof value !== 'boolean' ||
+                              typeof value !== 'string' ||
+                              isNaN(value));
+        if (nonValidType) {
```

Comment:
Isn't this missing the equivalent `env->options()->pending_deprecation` check? Otherwise we need to upgrade this from a "documentation-only" deprecation.

Reply:
Nope, this is inside [this conditional](https://github.com/nodejs/node/blob/910a555f2c6fb1442f6f0caa11d55aeed07d5e7f/lib/internal/bootstrap/pre_execution.js#L288), where `const pendingDeprecation = getOptionValue('--pending-deprecation');`, the equivalent of `env->options()->pending_deprecation`, based on my understanding 🤔